### PR TITLE
Make sure ID is ASCII

### PR DESCRIPTION
**Pull request by duglin**
Please only review the 2nd commit.

ping @jskeet

### CODE REVIEW

File: core/spec.md [hunk 1]
```diff
@@ -146,7 +146,8 @@ TODO: Add `format`
 - Type: String   # SHOULD this be a URI-Reference?
 - Description: An immutable unique identifier of the entity.
 - Constraints:
-  - MUST be a non-empty string
+  - MUST be a non-empty string that is compliant with the definition of
+    an HTTP header [`field-name`][field-name]`.
```
Comment (jskeet) [inline core/spec.md:150]: I believe this prohibits the use of `/` and `:` within an ID (as well as various other characters, of course). I can see those being pretty common within IDs... do we definitely want to prohibit them?

Comment (duglin) [reply core/spec.md:150]: oh wait - you're right, I messed up. I need to worry about valid http header values, not names. Doi! Will fix

File: core/spec.md [hunk 2]
```diff
@@ -146,7 +146,7 @@ TODO: Add `format`
 - Type: String   # SHOULD this be a URI-Reference?
 - Description: An immutable unique identifier of the entity.
 - Constraints:
-  - MUST be a non-empty string
+  - MUST be a non-empty string consisting of visible US-ASCII octets (33-126).
```
Comment (jskeet) [inline core/spec.md:149]: Assume we genuinely want all of those characters to be valid, that looks fine to me. I'm not in a position to judge whether we *do* want that character set :)

Comment (duglin) [reply core/spec.md:149]: I'm inclined to start with being less restrictive and wait until someone can identify a reason for us to ban more chars.

**Comment by duglin** [comment]
Approved on the 4/27 call

**Pull request merged**